File: 2015/day08/p02.py
#!/usr/bin/env python3
# Copyright (c) 2025 Adam Mullins
#
# Solution for Advent of Code 2015, Day 08, Part 2
#   https://adventofcode.com/2015/day/8#part2
from sys import stdin

orig_chars = 0
encoded_chars = 0
table = str.maketrans({'"': r"\"", "\\": "\\\\"})

for line in stdin:
    line = line.strip()
    orig_chars += len(line)
    encoded_chars += len(line.translate(table)) + 2

print(encoded_chars - orig_chars)

File: 2025/day01/p02.c
// Copyright (c) 2025 Adam Mullins
//
// Solution for Advent of Code 2025, Day 01, Part 02
// https://adventofcode.com/2025/day/1
#include <stdio.h>


#define LEN 2048
static char line[LEN];


int
main(void)
{
    char dir = '\0';
    int dist = 0;
    int pos = 50;
    int zero_count = 0; // Not to be confused with Count Zero.
    while (fgets(line, LEN, stdin)) {
        sscanf(&line[0], "%c%d", &dir, &dist);
        // Brute. Force.
        for(; dist > 0; dist--) {
            if (dir == 'L')
                pos == 0 ? pos = 99 : pos--;
            if (dir == 'R')
                pos == 99 ? pos = 0 : pos++;
            if (pos == 0)
                zero_count++;
        }
    }
    printf("%d\n", zero_count);
    return 0;
}
